specs/~dynamic-names.yml

```diff
@@ -0,0 +1,377 @@
+    data: { text: 'Hello, world!', example: 'partial'  }
+    template: '"{{>*example}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '"*Hello, world!*"'
+
+  - name: Dotted Names
+    desc: The dynamic partial should operate within the current context.
+    data: { text: 'Hello, world!', foo: { bar: { baz: 'partial' } } }
+    template: '"{{>*foo.bar.baz}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '"*Hello, world!*"'
+
+  - name: Dotted Names - Operator Precedence
+    desc: The dotted name should be resolved entirely before being dereferenced.
+    data:
+      text: 'Hello, world!'
+      foo: 'test'
+      test:
+        bar:
+          baz: 'partial'
+    template: '"{{>*foo.bar.baz}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '""'
+
+  - name: Dotted Names - Failed Lookup
+    desc: The dynamic partial should operate within the current context.
+    data:
+      foo:
+        text: 'Hello, world!'
+        bar:
+          baz: 'partial'
+    template: '"{{>*foo.bar.baz}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '"**"'
+
+  - name: Dotted names - Context Stacking
+    desc: Dotted names should not push a new frame on the context stack.
+    data:
+      section1: { value: 'section1' }
+      section2: { dynamic: 'partial', value: 'section2' }
+    template: "{{#section1}}{{>*section2.dynamic}}{{/section1}}"
+    partials:
+      partial: '"{{value}}"'
+    expected: '"section1"'
+
+  - name: Dotted names - Context Stacking Under Repetition
+    desc: Dotted names should not push a new frame on the context stack.
+    data:
+      value: 'test'
+      section1: [ 1, 2 ]
+      section2: { dynamic: 'partial', value: 'section2' }
+    template: "{{#section1}}{{>*section2.dynamic}}{{/section1}}"
+    partials:
+      partial: "{{value}}"
+    expected: "testtest"
+
+  - name: Dotted names - Context Stacking Failed Lookup
+    desc: Dotted names should resolve against the proper context stack.
+    data:
+      section1: [ 1, 2 ]
+      section2: { dynamic: 'partial', value: 'section2' }
+    template: "{{#section1}}{{>*section2.dynamic}}{{/section1}}"
+    partials:
+      partial: '"{{value}}"'
+    expected: '""""'
+
+  - name: Recursion
+    desc: Dynamic partials should properly recurse.
+    data:
+      template: 'node'
+      content: 'X'
+      nodes: [ { content: 'Y', nodes: [] } ]
+    template: '{{>*template}}'
+    partials: { node: '{{content}}<{{#nodes}}{{>*template}}{{/nodes}}>' }
+    expected: 'X<Y<>>'
+
+  - name: Dynamic Names - Double Dereferencing
+    desc: Dynamic Names can't be dereferenced more than once.
+    data: { dynamic: 'test', 'test': 'content' }
+    template: '"{{>**dynamic}}"'
+    partials: { content: 'Hello, world!' }
+    expected: '""'
+
+  - name: Dynamic Names - Composed Dereferencing
+    desc: Dotted Names are resolved entirely before dereferencing begins.
+    data: { foo: 'fizz', bar: 'buzz', fizz: { buzz: { 'content' } } }
```

Comment:
i don't know if it's a more recent addition to the spec, but the YAML parser I use for my spec test cases doesn't like implicit null values in JSON-like flow maps:

> Malformed inline YAML string { foo: 'fizz', bar: 'buzz', fizz: { buzz: { 'content' } } }

can we change this to the equivalent:

```yaml
data: { foo: 'fizz', bar: 'buzz', fizz: { buzz: { content: null } } }
```

Reply:
No problem, fortunately I had not tagged a release yet. Should be fixed now on master.